pkg/controller/migcluster/validation.go

```diff
@@ -82,6 +87,12 @@ func (r ReconcileMigCluster) validate(cluster *migapi.MigCluster) error {
 		return liberr.Wrap(err)
 	}
 
+	// cluster version
+	err = r.validateOperatorVersionMatchesHost(cluster)
+	if err != nil {
```

Comment:
@ibolton336 I think we need some modification here.

`validateOperatorVersionMatchesHost` will call
```
	clusterOperatorVersion, err := cluster.GetOperatorVersion(clusterClient)
	if err != nil {
		return liberr.Wrap(err)
	}
```

`cluster.GetOperatorVersion` includes
```
	operatorVersion, ok := clusterConfig.Data[OperatorVersionKey]
	if !ok {
		return "", liberr.Wrap(err)
	}
```

So I think `validateOperatorVersionMatchesHost` will return an error down the chain, which will cause the MigCluster to requeue. One of the ways an error gets generated will be if the configmap key isn't present, which we _are expecting to happen_. 

We should not pass the error back down and exit the reconcile for an _expected_ issue like this. Exiting a reconcile with an error will lead to a requeue, and in this case it would be an infinite requeue (not desired, can cause perf problems) until the user upgrades this MigCluster MTC install to a version including the expected configmap key.

```
	// Validations.
	err = r.validate(cluster)
	if err != nil {
		log.Trace(err)
		return reconcile.Result{Requeue: true}, nil
	}
```